### PR TITLE
Organizer Sign In Page

### LINKED ISSUE #70: Sign-in Page
Expose basic sign-in page for organizers. See https://next-auth.js.org/configuration/pages

* Make sure that it redirects to `/internal`. We can do this using getServerSideProps (see https://next-auth.js.org/configuration/pages#oauth-sign-in).
* Don't worry about design for this ticket (that's a later issue), just make it functional (you can even use the default nextauth page)

### PR DESCRIPTION
**Pull request by ArsalaanAli**
closes #70 

# Quick Overview of Changes

- Created styled oAuth button components for google and github
- Created sign in page for orgs
- Created a not-authorized modal that can be used to let users know they're being redirected, instead of just redirecting them without noticing them
- Added some tailwind colors for hw colours (will be adding all of them in the next ticket)

image

image

# Checklist

- [ ] If any frontend changes were made, include a screenshot/demo in the overview
- [ ] Checked all uses of changed component(s)/function(s)
- [ ] If any changes were made to our backend services, at least one (happy path) unit test was added OR a `TODO` comment was added to create one.
- [ ] Check that CI is passing.
- [ ] If no (re-)reviews after 1-2 days, ping the web leads on the discord to remind them to review the PR.

@basokant @ArsalaanAli

### CODE REVIEW

File: src/utils/redirect.ts
Comment (hunterchen7) [inline src/utils/redirect.ts:1]: don't organizers and applicants share the same login page?

Comment (ArsalaanAli) [reply src/utils/redirect.ts:1]: no

Comment (basokant) [reply src/utils/redirect.ts:1]: no

File: src/pages/internal/index.tsx [hunk 1]
```diff
@@ -51,6 +61,26 @@ function PreregistrationsButton() {
...
+    };
+  }
+  const user = await db.query.users.findFirst({
+    where: (users, { eq }) => eq(users.id, session.user.id),
+  });
+
+  return {
+    props: {
+      userSession: user,
+      name: "hello",
```
Comment (basokant) [inline src/pages/internal/index.tsx:81]: why is name here lol

File: src/pages/internal/login.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    });
+
+    return {
+      redirect: {
+        destination: "/internal",
+        permanent: false,
+      },
+    };
+  }
+  return { props: {} };
```
Comment (basokant) [inline src/pages/internal/login.tsx:56]: wait is it possible to just return {} here? w/o props

**Review by basokant — COMMENTED** [review]
nits lol

File: src/pages/internal/index.tsx (hunk 1 above)
Comment (ArsalaanAli) [reply src/pages/internal/index.tsx:81]: couldn't tell ya tbh

File: src/pages/internal/login.tsx (hunk 2 above)
Comment (ArsalaanAli) [reply src/pages/internal/login.tsx:56]: I get an error if i try to, so I'm assuming not

**Review by basokant — APPROVED** [review]
lgtm

**Pull request merged**